x/poe/keeper/legacy_distribution_querier.go

```diff
@@ -141,13 +141,20 @@ func (q legacyDistributionGRPCQuerier) DelegatorWithdrawAddress(c context.Contex
 	if req.DelegatorAddress == "" {
 		return nil, status.Error(codes.InvalidArgument, "delegator address cannot be empty")
 	}
-	add, err := sdk.AccAddressFromBech32(req.DelegatorAddress)
+	ownerAddr, err := sdk.AccAddressFromBech32(req.DelegatorAddress)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, "delegator address invalid")
+	}
 
+	// Query the `tg4-engagement` contract for the delegated withdraw address
+	ctx := sdk.UnwrapSDKContext(c)
+	gotVal, err := q.keeper.EngagementContract(ctx).QueryDelegated(ctx, ownerAddr)
```

Comment:
What do we return for an unknown address?

Reply:
For a valid address that had no delegations, the same address is returned. This is this case. What's missing is querying an address that actually has a delegation.